src/test/java/com/fabulousnesss/myproject/TicketServiceTest.java

```
@@ -0,0 +1,175 @@
+public class TicketServiceTest {
+    @Mock
+    private TicketRepository ticketRepository;
+
+    @InjectMocks
+    private TicketService ticketService;
+
+    @BeforeEach
+    public void init() {
+        ticketService = new TicketService(ticketRepository);
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideSaveTicketsData")
+    public void saveTickets_returnSavedTickets(List<Ticket> returnedTicketList, List<Ticket> expectedTicketList, int expectedSaves) {
+        if (expectedSaves > 0) {
+            Mockito.when(ticketRepository.saveAll(Mockito.anyList())).thenReturn(returnedTicketList);
+        }
+
+        List<Ticket> resultTickets = ticketService.saveTickets(returnedTicketList);
+
+        Assertions.assertEquals(expectedTicketList.size(), resultTickets.size());
+        for (int i = 0; i < expectedTicketList.size(); i++) {
+            Ticket expectedTicket = expectedTicketList.get(i);
+            Ticket actualTicket = resultTickets.get(i);
+            Assertions.assertEquals(expectedTicket.getId(), actualTicket.getId());
+            Assertions.assertEquals(expectedTicket.getUser().getId(), actualTicket.getUser().getId());
+        }
+        Mockito.verify(ticketRepository, Mockito.times(expectedSaves)).saveAll(Mockito.anyList());
+    }
+
+    private static Stream<Arguments> provideSaveTicketsData() {
+        Ticket ticketWithExistingUser = new Ticket(1L);
+        ticketWithExistingUser.setUser(new User(1L));
+
+        Ticket ticketWithoutExistingUser = new Ticket(-1L);
+        ticketWithExistingUser.setUser(new User(-1L));
+
+        Ticket ticketAndUserWithZeroId = new Ticket(0L);
+        ticketWithExistingUser.setUser(new User(0L));
+
+        return Stream.of(
+                Arguments.of(Collections.singletonList(ticketWithExistingUser), Collections.singletonList(ticketWithExistingUser), 1),
+                Arguments.of(Collections.singletonList(ticketWithoutExistingUser), Collections.emptyList(), 0),
```

Probably in this case we should also throw an exception?
And in this case would be better to have separate tests without combining it with ParameterizedTest